interpreter/decorators.go

```diff
@@ -177,25 +177,51 @@ func maybeOptimizeSetMembership(i Interpretable, inlist InterpretableCall) (Inte
 		return NewConstValue(inlist.ID(), types.False), nil
 	}
 	it := list.Iterator()
-	var typ ref.Type
 	valueSet := make(map[ref.Val]ref.Val)
 	for it.HasNext() == types.True {
 		elem := it.Next()
 		if !types.IsPrimitiveType(elem) {
 			// Note, non-primitive type are not yet supported.
 			return i, nil
 		}
-		if typ == nil {
-			typ = elem.Type()
-		} else if typ.TypeName() != elem.Type().TypeName() {
-			return i, nil
-		}
 		valueSet[elem] = types.True
+		switch ev := elem.(type) {
```

Comment:
Hmmmm. Why are we only checking for lossless conversions for types.Double but not the others?

Reply:
The others are getting overflow checks as part of the `ConvertToType` call, but the double conversion to an int or uint in this block must only be permitted when the double is int-like; hence, the additional check.

Reply:
Discussed offline.